go/vt/vttablet/tabletserver/tabletenv/config.go

```diff
@@ -879,6 +888,48 @@ func (c *TabletConfig) Verify() error {
 	return nil
 }
 
+// verifyUnmanagedTabletConfig checks unmanaged tablet related config for sanity
+func (c *TabletConfig) verifyUnmanagedTabletConfig() error {
+	isUnmanaged := c.UnmanagedTablet
+
+	// Skip checks if tablet is not unmanaged
+	if !isUnmanaged {
+		return nil
+	}
+
+	// Throw error if both host and socket are null
+	if !c.DB.HasGlobalSettings() {
+		return errors.New("no connection parameters specified but unmanaged mode specified")
+	}
+	if c.DB.App.User == "" {
+		return errors.New("database app user not specified")
+	}
+	if c.DB.App.Password == "" {
+		return errors.New("database app user password not specified")
+	}
+
+	return c.checkConnectionForExternalMysql()
+}
+
+// Test connectivity of external mysql
+func (c *TabletConfig) checkConnectionForExternalMysql() error {
+	params := mysql.ConnParams{
+		Host:       c.DB.Host,
+		Port:       c.DB.Port,
+		DbName:     c.DB.DBName,
+		Uname:      c.DB.App.User,
+		Pass:       c.DB.App.Password,
+		UnixSocket: c.DB.Socket,
+	}
+
+	conn, err := mysql.Connect(context.Background(), &params)
+	if err != nil {
+		return err
+	}
```

Comment:
Whenever a conn is created, and the error check has passed, we should always close the connection in a defer call to prevent connections from leaking.
`defer conn.Close()` should be added after this code.

Reply:
got it, done.